File: Java/src/main/java/sofe/exercise3/MaxPathTreeCount.java
package sofe.exercise3;

public class MaxPathTreeCount {

    public static int maxPathCount(Node root){
        return 0;
    }

}

class Node{
    Node left;
    Node right;
    int value;

    public Node(int value){
        this.value = value;
    }

    public Node(Node left, Node right, int value){
        this.left = left;
        this.right = right;
        this.value = value;
    }
}

File: C Sharp/Sofe/Exercise3/MaxPathTreeCount.cs
using System;

namespace Sofe.Exercise3
{
    public class MaxPathTreeCount
    {
        internal static int MaxPathCount(Node Root)
        {
            return 0;
        }
    }

}

class Node
{
    Node Left;
    Node Right;
    int Value;

    public Node(int Value, Node Left = null, Node Right = null)
    {
        this.Left = Left;
        this.Right = Right;
        this.Value = Value;
    }
}